<?xml version="1.0"?>
<launch>

  <node name="handsfree_hw_node" pkg="handsfree_hw" type="handsfree_hw_node" />

  <param name="robot_description" textfile="$(find XM)/robots/XM.URDF" />
  
<!-- load the re controllers -->
  <rosparam file="$(find handsfree_hw)/config/base_controller.yaml" command="load"/>

  <!-- load the controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" ns="/mobile_base" args=" mobile_base_controller"/>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher">
  <remap from="/joint_states" to="/mobile_base/joint_states"/>
  </node>
	
	<node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager"/>
  	<node pkg="nodelet" type="nodelet" name="velocity_smoother" 
	    args="load xm_velocity_smoother/VelocitySmootherNodelet nodelet_manager" >
    <rosparam file="$(find xm_velocity_smoother)/param/standalone.yaml" command="load"/>
	  <remap from="velocity_smoother/raw_cmd_vel" to="/mobile_base/mobile_base_controller/cmd_vel"/>
      <remap from="velocity_smoother/smooth_cmd_vel" to="/mobile_base/mobile_base_controller/smooth_cmd_vel"/>
      <remap from="/velocity_smoother/odometry" to="/mobile_base/mobile_base_controller/odom"/>
   </node>
</launch>
